.github/workflows/ci-windows-shared-local.yml

```diff
@@ -0,0 +1,59 @@
+# Do a shared library build with local dependencies.
+
+name: CI Windows Shared Local
+on: [push, pull_request]
+
+permissions:
+  contents: read
+
+# Cancel the workflow if a new one is triggered from the same PR, branch, or tag, except on main.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  build-windows-shared-local:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler: [msvc, clang-cl]
+
+    steps:
+    - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+    - uses: ./.github/actions/setup-windows
+      with:
+        codec-aom: 'LOCAL'
+        codec-dav1d: 'LOCAL'
+        extra-cache-key: ${{ matrix.compiler }}
+    - name: Leave compiler as default
+      if: matrix.compiler == 'msvc'
+      run: |
+        echo "AVIF_CMAKE_C_COMPILER=" >> $env:GITHUB_ENV
+        echo "AVIF_CMAKE_CXX_COMPILER=" >> $env:GITHUB_ENV
+    - name: Set clang-cl as compiler
+      if: matrix.compiler == 'clang-cl'
+      run: |
+        echo "AVIF_CMAKE_C_COMPILER=-DCMAKE_C_COMPILER=clang-cl" >> $env:GITHUB_ENV
+        echo "AVIF_CMAKE_CXX_COMPILER=-DCMAKE_CXX_COMPILER=clang-cl" >> $env:GITHUB_ENV
+
+    - name: Prepare libavif (cmake)
+      run: >
+        cmake -G Ninja -S . -B build
+        -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON
```

Comment:
Should `Debug` be tested too? I remember difficulties when linking differently built dependencies and wonder if that should be exercised. Probably not worth the effort.

Reply:
We can looking into adding `Debug` later. Right now I am focusing on getting a clean (no compiler or linker warnings) `Release` build.